### ISSUE TITLE
Test failing but not failing action 

### ISSUE THREAD

**Issue by paigerube14** [redhat-chaos/krkn#453]
In the build action in pr's a test is failing but it is not failing the action overall 

https://github.com/redhat-chaos/krkn/actions/runs/5624388976

```
Test | Result | Duration
-- | -- | --
test_net_chaos | Fail | 0:0:9
```

**Comment by Transcendental-Programmer** [comment]
Hi, I’ve started looking into this issue. It looks like the file .github/workflows/tests.yml is the one that runs and handles the test results. I think changing this file can fix the problem.

Let me know if it’s okay for me to make the changes or if there’s anything else I should know first.

**Comment by Transcendental-Programmer** [comment]
Hey @paigerube14 !

I found two problems with the GitHub Actions workflow that I fixed in the tests.yaml file:

## problem 1: Tests failing but CI passing anyways

I saw that when tests failed, the CI workflow was still showing green (success). 
the error was at line 155

```
        run: grep Fail CI/results.markdown && false || true
```

so i tested 

```
$ echo "test_net_chaos | Fail | 0:0:9" > test_results.markdown
$ grep Fail test_results.markdown && false || true
test_net_chaos | Fail | 0:0:9
$ echo $?
0
```

This test shows whats going on - the old command always gives exit code 0 (success) even when it finds failures! This happens because the `|| true` part at the end makes the whole command success no matter what. Thats why CI was passing even with failing tests.

Then I tried the new way:
```
$ ! grep Fail test_results.markdown
test_net_chaos | Fail | 0:0:9
$ echo $?
1
```

Now it gives exit code 1 (failure) when it finds test failures - which is what we want! The `!` at the start flips the result, so finding "Fail" makes the whole command fail.

## Problem 2: Empty Git email 

i also found another bug.
I saw the email for git commits was just empty (`<>`). This can break things when pushing changes.

I looked at GitHub's docs about [commit emails](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses) check the note 


Git needs a real email format for commits. So I changed it to:
```yaml
git config user.email "krkn-actions@users.noreply.github.com"
```

This follows how GitHub usually does emails for automated stuff. If you want a different email for these automated commits, just let me know!

Does this look good to you? I tested both changes and they work, but wanted to check with you!

[my commit](https://github.com/krkn-chaos/krkn/commit/be316dc3972f41f270520a34b577b8aa3fc2ccd0)

Thanks!

**Comment by Transcendental-Programmer** [comment]
should i create a pr for this?
does this change require documentation if yes tell me i will make those (i dont feel like it does)

**Comment by paigerube14** [comment]
@Transcendental-Programmer yes! if you have an idea of how to update this would greatly appreciate a PR! This is also a pretty old issue, not sure if it is still relevant!
